src/Module.cpp

```diff
@@ -199,19 +200,91 @@ static Registerer registerer;
     }
 }
 
+std::string indent_string(const std::string &src, const std::string &indent) {
+    std::ostringstream o;
+    o << indent;
+    for (size_t i = 0; i < src.size(); i++) {
+        const char c = src[i];
+        o << c;
+        if (c == '\n' && i != src.size() - 1) {
+            o << indent;
+        }
+    }
+    return o.str();
+}
+
+void emit_schedule_file(const std::string &name,
+                        const std::vector<Target> &targets,
+                        const std::string &scheduler_name,
+                        const std::string &machine_params_string,
+                        const std::string &body,
+                        std::ostream &stream) {
+    std::string s = R"INLINE_CODE(
+#ifndef $CLEANNAME$_SCHEDULE_H
+#define $CLEANNAME$_SCHEDULE_H
+
+// MACHINE GENERATED -- DO NOT EDIT
+// This schedule was automatically generated by $SCHEDULER$
+// for target=$TARGET$
+// with machine_params=$MACHINEPARAMS$
+
+#include "Halide.h"
+
+$NSOPEN$
```

Comment:
nit: had to think about this too long. NS->NAMESPACE

Reply:
done